functional_tests.go

```diff
@@ -255,20 +287,21 @@ func testPutObjectReadAt() {
 
 	// Generate data using 4 parts so that all 3 'workers' are utilized and a part is leftover.
 	// Use different data for each part for multipart tests to ensure part order at the end.
-	var buf = getDataBuffer("datafile-65-MB", MinPartSize)
+	var reader = getDataReader("datafile-65-MB", MinPartSize)
+	defer reader.Close()
 
 	// Save the data
 	objectName := randString(60, rand.NewSource(time.Now().UnixNano()), "")
 	// Object content type
 	objectContentType := "binary/octet-stream"
 
-	n, err := c.PutObject(bucketName, objectName, bytes.NewReader(buf), objectContentType)
+	n, err := c.PutObject(bucketName, objectName, reader, objectContentType)
 	if err != nil {
 		logger().Fatal("Error:", err, bucketName, objectName)
 	}
 
-	if n != int64(len(buf)) {
-		logger().Fatalf("Error: number of bytes does not match, want %v, got %v\n", len(buf), n)
+	if n != int64(MinPartSize) {
```

Comment:
If we are not using a random data source, then we get our data from 'datafile-65-MB' which means 65MB and not MinPartSize (64MB)